DataStructures/ArrayBasedCircularQueue/Enumerators/QueueEnumerator.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+
+namespace DataStructures.ArrayBasedCircularQueue.Enumerators
+{
+    /// <summary>
+    /// Queue Enumerator to enumerate queue.
+    /// </summary>
+    /// <typeparam name="T">Generic Type.</typeparam>
+    public class QueueEnumerator<T> : IEnumerator<T>
```

Comment:
Codacy found an issue: [Fix this implementation of 'IDisposable' to conform to the dispose pattern.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6301193)